bindings/mysql/mysql.go

```diff
@@ -285,84 +287,54 @@ func jsonify(rows *sql.Rows) ([]byte, error) {
 
 	var ret []interface{}
 	for rows.Next() {
-		scanArgs := prepareScanArgs(columnTypes)
-		err := rows.Scan(scanArgs...)
+		values := prepareValues(columnTypes)
+		err := rows.Scan(values...)
 		if err != nil {
 			return nil, err
 		}
 
-		r := convertScanArgs(columnTypes, scanArgs)
+		r := convert(columnTypes, values)
 		ret = append(ret, r)
 	}
 
 	return json.Marshal(ret)
 }
 
-func convertScanArgs(columnTypes []*sql.ColumnType, scanArgs []interface{}) map[string]interface{} {
-	r := map[string]interface{}{}
-
-	for i, v := range columnTypes {
-		if s, ok := (scanArgs[i]).(*sql.NullString); ok {
-			r[v.Name()] = s.String
-
-			continue
-		}
-
-		if s, ok := (scanArgs[i]).(*sql.NullBool); ok {
-			r[v.Name()] = s.Bool
-
-			continue
-		}
-
-		if s, ok := (scanArgs[i]).(*sql.NullInt32); ok {
-			r[v.Name()] = s.Int32
-
-			continue
-		}
-
-		if s, ok := (scanArgs[i]).(*sql.NullInt64); ok {
-			r[v.Name()] = s.Int64
-
-			continue
-		}
+func prepareValues(columnTypes []*sql.ColumnType) []interface{} {
+	types := make([]reflect.Type, len(columnTypes))
+	for i, tp := range columnTypes {
+		types[i] = tp.ScanType()
+	}
 
-		if s, ok := (scanArgs[i]).(*sql.NullFloat64); ok {
-			r[v.Name()] = s.Float64
+	values := make([]interface{}, len(columnTypes))
+	for i := range values {
+		values[i] = reflect.New(types[i]).Interface()
+	}
 
-			continue
-		}
+	return values
+}
 
-		if s, ok := (scanArgs[i]).(*sql.NullTime); ok {
-			r[v.Name()] = s.Time
+func convert(columnTypes []*sql.ColumnType, values []interface{}) map[string]interface{} {
+	r := map[string]interface{}{}
 
-			continue
+	for i, ct := range columnTypes {
+		value := values[i]
+
+		switch v := values[i].(type) {
```

Comment:
How are null values handled? Is there a case needed for those to explicitly set the key to `nil`?

Reply:
I revised it to 

```go
		if value != nil {
			r[ct.Name()] = value
		}
```